File: Beecrowd/Beginner./1049.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string w1, w2, w3;
    cin >> w1 >> w2 >> w3;

    if (w1 == "vertebrado") {
        if (w2 == "ave") {
            if (w3 == "carnivoro")
                cout << "aguia" << endl;
            else if (w3 == "onivoro")
                cout << "pomba" << endl;
        } else if (w2 == "mamifero") {
            if (w3 == "onivoro")
                cout << "homem" << endl;
            else if (w3 == "herbivoro")
                cout << "vaca" << endl;
        }
    } else if (w1 == "invertebrado") {
        if (w2 == "inseto") {
            if (w3 == "hematofago")
                cout << "pulga" << endl;
            else if (w3 == "herbivoro")
                cout << "lagarta" << endl;
        } else if (w2 == "anelideo") {
            if (w3 == "hematofago")
                cout << "sanguessuga" << endl;
            else if (w3 == "onivoro")
                cout << "minhoca" << endl;
        }
    }

    return 0;
}

File: Beecrowd/1. Beginner/1049/1049.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        String w1 = input.nextLine();
        String w2 = input.nextLine();
        String w3 = input.nextLine();

        if (w1.equals("vertebrado")) {
            if (w2.equals("ave")) {
                if (w3.equals("carnivoro")) {
                    System.out.println("aguia");
                } else if (w3.equals("onivoro")) {
                    System.out.println("pomba");
                }
            } else if (w2.equals("mamifero")) {
                if (w3.equals("onivoro")) {
                    System.out.println("homem");
                } else if (w3.equals("herbivoro")) {
                    System.out.println("vaca");
                }
            }
        } else if (w1.equals("invertebrado")) {
            if (w2.equals("inseto")) {
                if (w3.equals("hematofago")) {
                    System.out.println("pulga");
                } else if (w3.equals("herbivoro")) {
                    System.out.println("lagarta");
                }
            } else if (w2.equals("anelideo")) {
                if (w3.equals("hematofago")) {
                    System.out.println("sanguessuga");
                } else if (w3.equals("onivoro")) {
                    System.out.println("minhoca");
                }
            }
        }

        input.close();
    }
}
